<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>vartrix | Single-Cell Genotyping Tool | anno-sc</title>
  <meta name="keywords" content=" singlecell , SNP ">
  <meta name="description" content="vartrix | Single-Cell Genotyping Tool | anno-sc">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="description" content="VarTrix是一款10XGenomics官方的软件用于从10XGenomics单细胞数据中提取单细胞变异信息。  VarTrix 并不执行variant calling 需要输入先前定义的variant calls VarTrix可用于评估样品中单细胞的异质性  VarTrix 流程图 githubhttps:&#x2F;&#x2F;github.com&#x2F;10XGenomics&#x2F;vartrix 下载安装VarTri">
<meta property="og:type" content="article">
<meta property="og:title" content="vartrix | Single-Cell Genotyping Tool">
<meta property="og:url" content="https://anno-sc.com/2020/11/30/vartrix/index.html">
<meta property="og:site_name" content="anno-sc">
<meta property="og:description" content="VarTrix是一款10XGenomics官方的软件用于从10XGenomics单细胞数据中提取单细胞变异信息。  VarTrix 并不执行variant calling 需要输入先前定义的variant calls VarTrix可用于评估样品中单细胞的异质性  VarTrix 流程图 githubhttps:&#x2F;&#x2F;github.com&#x2F;10XGenomics&#x2F;vartrix 下载安装VarTri">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_1.png">
<meta property="og:image" content="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_2.png">
<meta property="og:image" content="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_3.png">
<meta property="article:published_time" content="2020-11-30T08:28:00.000Z">
<meta property="article:modified_time" content="2020-11-30T09:07:34.660Z">
<meta property="article:author" content="anno-sc">
<meta property="article:tag" content="singlecell">
<meta property="article:tag" content="SNP">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_1.png">


<link rel="icon" href="/img/avatar.jpg">

<link href="/css/style.css?v=1.1.0" rel="stylesheet">

<link href="/css/hl_theme/atom-light.css?v=1.1.0" rel="stylesheet">

<link href="//cdn.jsdelivr.net/npm/animate.css@4.1.0/animate.min.css" rel="stylesheet">

<script src="//cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js"></script>
<script src="/js/titleTip.js?v=1.1.0" ></script>

<script src="//cdn.jsdelivr.net/npm/highlightjs@9.16.2/highlight.pack.min.js"></script>
<script>
    hljs.initHighlightingOnLoad();
</script>

<script src="//cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js"></script>



<script src="//cdn.jsdelivr.net/npm/jquery.cookie@1.4.1/jquery.cookie.min.js" ></script>

<script src="/js/iconfont.js?v=1.1.0" ></script>

<meta name="generator" content="Hexo 5.2.0"></head>
<div style="display: none">
  <input class="theme_disqus_on" value="false">
  <input class="theme_preload_comment" value="">
  <input class="theme_blog_path" value="">
  <input id="theme_shortcut" value="true" />
  <input id="theme_highlight_on" value="true" />
  <input id="theme_code_copy" value="true" />
</div>



<body>
<aside class="nav">
    <div class="nav-left">
        <a href="/" class="avatar_target">
    <img class="avatar" src="/img/avatar.jpg" />
</a>
<div class="author">
    <span>anno-sc</span>
</div>

<div class="icon">
    
        
        <a title="github" href="https://github.com/anno-sc" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-github"></use>
                </svg>
            
        </a>
        
    
</div>




<ul>
    <li><div class="all active" data-rel="全部文章">全部文章<small>(7)</small></div></li>
    
        
            
            <li><div data-rel="QC">QC<small>(2)</small></div>
                
            </li>
            
        
    
        
            
            <li><div data-rel="plot">plot<small>(1)</small></div>
                
            </li>
            
        
    
        
            
            <li><div data-rel="SNP">SNP<small>(1)</small></div>
                
            </li>
            
        
    
        
            
            <li><div data-rel="VDJ">VDJ<small>(1)</small></div>
                
            </li>
            
        
    
        
            
            <li><div data-rel="免疫细胞">免疫细胞<small>(1)</small></div>
                
            </li>
            
        
    
        
            
            <li><div data-rel="生产力工具">生产力工具<small>(1)</small></div>
                
            </li>
            
        
    
</ul>
<div class="left-bottom">
    <div class="menus">
    
    
    
    
    </div>
    <div></div>
</div>
<input type="hidden" id="yelog_site_posts_number" value="7">

<div style="display: none">
    <span id="busuanzi_value_site_uv"></span>
    <span id="busuanzi_value_site_pv"></span>
</div>

    </div>
    <div class="nav-right">
        <div class="friends-area">
    <div class="friends-title">
        友情链接
        <i class="iconfont icon-left"></i>
    </div>
    <div class="friends-content">
        <ul>
            
            <li><a target="_blank" href="http://seqyuan.com">seqyuan</a></li>
            
        </ul>
    </div>
</div>
        <div class="title-list">
    <div class="right-top">
        <div id="default-panel">
            <i class="iconfont icon-search" data-title="搜索 快捷键 i"></i>
            <div class="right-title">全部文章</div>
            <i class="iconfont icon-file-tree" data-title="切换到大纲视图 快捷键 w"></i>
        </div>
        <div id="search-panel">
            <i class="iconfont icon-left" data-title="返回"></i>
            <input id="local-search-input" />
            <label class="border-line" for="input"></label>
            <i class="iconfont icon-case-sensitive" data-title="大小写敏感"></i>
            <i class="iconfont icon-tag" data-title="标签"></i>
        </div>
        <div id="outline-panel" style="display: none">
            <div class="right-title">大纲</div>
            <i class="iconfont icon-list" data-title="切换到文章列表"></i>
        </div>
    </div>

    <div class="tags-list">
    <input id="tag-search" />
    <div class="tag-wrapper">
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>cellranger</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>DC</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>NK</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>Rmarkdown</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>singlecell</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>SNP</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>T细胞</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>VDJ</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>VlnPlot</a>
            </li>
        
            <li class="article-tag-list-item">
                <i class="iconfont icon-tag"></i><a>双细胞</a>
            </li>
        
    </div>

</div>

    
    <nav id="title-list-nav">
        
        <a  class="全部文章 SNP "
           href="/2020/11/30/vartrix/"
           data-tag="singlecell,SNP"
           data-author="ahworld07" >
            <span class="post-title" title="vartrix | Single-Cell Genotyping Tool">vartrix | Single-Cell Genotyping Tool</span>
            <span class="post-date" title="2020-11-30 16:28:00">2020/11/30</span>
        </a>
        
        <a  class="全部文章 VDJ "
           href="/2020/09/27/T%E7%BB%86%E8%83%9EVDJ%E7%9A%84TCR_dist%E5%88%86%E6%9E%90%E7%BB%93%E6%9E%9C%E8%AF%B4%E6%98%8E/"
           data-tag="T细胞,VDJ"
           data-author="尧小飞" >
            <span class="post-title" title="T细胞VDJ的TCR_dist分析结果说明">T细胞VDJ的TCR_dist分析结果说明</span>
            <span class="post-date" title="2020-09-27 14:54:00">2020/09/27</span>
        </a>
        
        <a  class="全部文章 免疫细胞 "
           href="/2020/09/25/NK_DC_lineage/"
           data-tag="DC,NK"
           data-author="Spartanzhao" >
            <span class="post-title" title="DC和NK细胞的lineage及marker基因">DC和NK细胞的lineage及marker基因</span>
            <span class="post-date" title="2020-09-25 16:00:00">2020/09/25</span>
        </a>
        
        <a  class="全部文章 QC "
           href="/2020/09/24/doublets/"
           data-tag="singlecell,双细胞"
           data-author="ahworld07" >
            <span class="post-title" title="单细胞质控-双细胞鉴定软件">单细胞质控-双细胞鉴定软件</span>
            <span class="post-date" title="2020-09-24 16:51:00">2020/09/24</span>
        </a>
        
        <a  class="全部文章 plot "
           href="/2020/08/04/R%E5%87%BD%E6%95%B0%E5%AE%9E%E7%8E%B0%E5%8D%95%E7%BB%86StackedVlnPlot/"
           data-tag="VlnPlot"
           data-author="ahworld07" >
            <span class="post-title" title="R函数实现单细胞StackedVlnPlot">R函数实现单细胞StackedVlnPlot</span>
            <span class="post-date" title="2020-08-04 13:38:00">2020/08/04</span>
        </a>
        
        <a  class="全部文章 QC "
           href="/2020/07/12/20200712-%E5%A7%9A%E7%9B%9F%E6%88%90-cellranger4.0%E6%8A%A2%E5%85%88%E7%9C%8B/"
           data-tag="cellranger"
           data-author="尧小飞" >
            <span class="post-title" title="cellranger4.0抢先看">cellranger4.0抢先看</span>
            <span class="post-date" title="2020-07-12 14:36:00">2020/07/12</span>
        </a>
        
        <a  class="全部文章 生产力工具 "
           href="/2020/07/01/20200701-%E5%A7%9A%E7%9B%9F%E6%88%90-Rmarkdown%E7%94%9F%E6%88%90%E7%BD%91%E9%A1%B5%E7%89%88%E7%BB%93%E9%A2%98%E6%8A%A5%E5%91%8A%E5%AE%8C%E7%BE%8E%E6%96%B9%E6%A1%88/"
           data-tag="Rmarkdown"
           data-author="尧小飞" >
            <span class="post-title" title="Rmarkdown生成网页版结题报告完美方案">Rmarkdown生成网页版结题报告完美方案</span>
            <span class="post-date" title="2020-07-01 14:54:00">2020/07/01</span>
        </a>
        
        <div id="no-item-tips">

        </div>
    </nav>
    <div id="outline-list">
    </div>
</div>
    </div>
    <div class="hide-list">
        <div class="semicircle" data-title="切换全屏 快捷键 s">
            <div class="brackets first"><</div>
            <div class="brackets">&gt;</div>
        </div>
    </div>
</aside>
<div id="post">
    <div class="pjax">
        <article id="post-vartrix" class="article article-type-post" itemscope itemprop="blogPost">
    
        <h1 class="article-title">vartrix | Single-Cell Genotyping Tool</h1>
    
    <div class="article-meta">
        
        
        <span class="author"><a>ahworld07</a></span>
        
        
        <span class="book">
            <i class="iconfont icon-category"></i>
            
            
            <a  data-rel="SNP">SNP</a>
            
        </span>
        
        
        <span class="tag">
            <i class="iconfont icon-tag"></i>
            
            <a class="color1">singlecell</a>
            
            <a class="color4">SNP</a>
            
        </span>
        
    </div>
    <div class="article-meta">
        
            发布时间 : <time class="date" title='最后更新: 2020-11-30 17:07:34'>2020-11-30 16:28</time>
        
    </div>
    <div class="article-meta">
        
        
        <span id="busuanzi_container_page_pv">
            阅读 :<span id="busuanzi_value_page_pv">
                <span class="count-comment">
                    <span class="spinner">
                      <div class="cube1"></div>
                      <div class="cube2"></div>
                    </span>
                </span>
            </span>
        </span>
        
        
        <span class="top-comment" title="跳转至评论区">
            <a href="#comments">
                评论:<span class="count-comment">
                    <span class="spinner">
                      <div class="cube1"></div>
                      <div class="cube2"></div>
                    </span>
                </span>
            </a>
        </span>
        
    </div>
    
    <div class="toc-ref">
    
        <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#VarTrix-%E6%B5%81%E7%A8%8B%E5%9B%BE"><span class="toc-text">VarTrix 流程图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#github"><span class="toc-text">github</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%B8%8B%E8%BD%BD%E5%AE%89%E8%A3%85"><span class="toc-text">下载安装</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%B5%8B%E8%AF%95"><span class="toc-text">测试</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%B8%8B%E8%BD%BD%E6%B5%8B%E8%AF%95%E6%95%B0%E6%8D%AE"><span class="toc-text">下载测试数据</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BD%BF%E7%94%A8%E8%AF%B4%E6%98%8E"><span class="toc-text">使用说明</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%B5%8B%E8%AF%95%E4%BB%A3%E7%A0%81"><span class="toc-text">测试代码</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#UMAP%E5%9B%BE%E4%B8%AD%E5%B1%95%E7%A4%BAVarTrix%E7%BB%93%E6%9E%9C"><span class="toc-text">UMAP图中展示VarTrix结果</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#VarTrix%E7%BB%93%E6%9E%9C%E8%BD%AC%E4%B8%BA%E5%8F%AF%E6%B7%BB%E5%8A%A0%E7%BB%99Seurat%E5%AF%B9%E8%B1%A1%E7%9A%84MetaData"><span class="toc-text">VarTrix结果转为可添加给Seurat对象的MetaData</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#%E8%AF%BB%E5%8F%96VarTrix%E7%9A%84%E7%BB%93%E6%9E%9Cmatrix-barcodes%E5%92%8Cvariants"><span class="toc-text">读取VarTrix的结果matrix, barcodes和variants</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#%E6%B7%BB%E5%8A%A0VarTrix%E4%BF%A1%E6%81%AF%E7%BB%99Seurat%E5%AF%B9%E8%B1%A1"><span class="toc-text">添加VarTrix信息给Seurat对象</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#UMAP%E5%B1%95%E7%A4%BA%E7%BB%93%E6%9E%9C"><span class="toc-text">UMAP展示结果</span></a></li></ol></li></ol></li></ol>
    
<style>
    .left-col .switch-btn,
    .left-col .switch-area {
        display: none;
    }
    .toc-level-3 i,
    .toc-level-3 ol {
        display: none !important;
    }
</style>
</div>
    
    <div class="article-entry" itemprop="articleBody">
      
        <p>VarTrix是一款10XGenomics官方的软件用于从10XGenomics单细胞数据中提取单细胞变异信息。</p>
<ul>
<li>VarTrix 并不执行<code>variant calling</code></li>
<li>需要输入先前定义的<code>variant calls</code></li>
<li>VarTrix可用于评估样品中单细胞的异质性</li>
</ul>
<h2 id="VarTrix-流程图"><a href="#VarTrix-流程图" class="headerlink" title="VarTrix 流程图"></a>VarTrix 流程图</h2><p><img src="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_1.png"></p>
<h2 id="github"><a href="#github" class="headerlink" title="github"></a>github</h2><p><a target="_blank" rel="noopener" href="https://github.com/10XGenomics/vartrix">https://github.com/10XGenomics/vartrix</a></p>
<h2 id="下载安装"><a href="#下载安装" class="headerlink" title="下载安装"></a>下载安装</h2><p>VarTrix使用rust语言编写，官方提供了编译后的<a target="_blank" rel="noopener" href="https://github.com/10XGenomics/vartrix/releases/tag/v1.1.16">二进制文件</a>，直接下载就能使用。</p>
<p><img src="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_2.png"></p>
<p>我们以linux版为例，下载后增加可执行权限。</p>
<pre><code class="shell">chmod +x vartrix_linux</code></pre>
<h2 id="测试"><a href="#测试" class="headerlink" title="测试"></a>测试</h2><h3 id="下载测试数据"><a href="#下载测试数据" class="headerlink" title="下载测试数据"></a>下载测试数据</h3><p>为了快速理解VarTrix的使用，我们从github下载其测试数据<br><a target="_blank" rel="noopener" href="https://github.com/10XGenomics/vartrix/test">https://github.com/10XGenomics/vartrix/test</a></p>
<p>test目录里提供了多套测试数据，我们选下面这一组输入文件做测试：</p>
<ul>
<li>test_dna.vcf<br>单样品的VCF格式变异文件，用于输入</li>
<li>test_dna.bam，test_dna.bam.bai<br>Input Cell Ranger BAM，需要有index文件</li>
<li>test_dna.fa, test_dna.fa.fai<br>FASTA格式的参考基因组文件，需要有index文件</li>
<li>dna_barcodes.tsv<br>Cell Ranger输出的矩阵barcode文件</li>
</ul>
<h3 id="使用说明"><a href="#使用说明" class="headerlink" title="使用说明"></a>使用说明</h3><p><code>./vartrix_linux -h</code>查看程序使用说明</p>
<pre><code>vartrix 1.1.14
Ian Fiddes &lt;ian.fiddes@10xgenomics.com&gt; and Patrick Marks &lt;patrick@10xgenomics.com&gt;
Variant assignment for single cell genomics

USAGE:
    vartrix_linux [FLAGS] [OPTIONS] --bam &lt;FILE&gt; --cell-barcodes &lt;FILE&gt; --fasta &lt;FILE&gt; --vcf &lt;FILE&gt;

FLAGS:
    -h, --help                  Prints help information
        --no-duplicates         Do not consider duplicate alignments
        --primary-alignments    Use primary alignments only
        --umi                   Consider UMI information when populating coverage matrices?
    -V, --version               Prints version information

OPTIONS:
    -b, --bam &lt;FILE&gt;                         Cellranger BAM file
        --bam-tag &lt;bam_tag&gt;                  BAM tag to consider for marking cells? [default: CB]
    -c, --cell-barcodes &lt;FILE&gt;               File with cell barcodes to be evaluated
    -f, --fasta &lt;FILE&gt;                       Genome fasta file
        --log-level &lt;log_level&gt;              Logging level [default: error]  [possible values: info, debug, error]
        --mapq &lt;INTEGER&gt;                     Minimum read mapping quality to consider [default: 0]
    -o, --out-matrix &lt;OUTPUT_FILE&gt;           Output Matrix Market file (.mtx) [default: out_matrix.mtx]
        --out-variants &lt;OUTPUT_FILE&gt;         Output variant file. Reports ordered list of variants to help with loading into
                                             downstream tools
    -p, --padding &lt;INTEGER&gt;                  Number of padding to use on both sides of the variant. Should be at least 1/2 of read
                                             length [default: 100]
        --ref-matrix &lt;OUTPUT_FILE&gt;           Location to write reference Matrix Market file. Only used if --scoring-method is
                                             coverage [default: ref_matrix.mtx]
    -s, --scoring-method &lt;scoring_method&gt;    Type of matrix to produce. In &#39;consensus&#39; mode, cells with both ref and alt reads are
                                             given a 3, alt only reads a 2, and ref only reads a 1. Suitable for clustering.  In
                                             &#39;coverage&#39; mode, it is required that you set --ref-matrix to store the second matrix
                                             in. The &#39;alt_frac&#39; mode will report the fraction of alt reads, which is effectively the
                                             ratio of the alternate matrix to the sum of the alternate and coverage matrices.
                                             [default: consensus]  [possible values: consensus, coverage, alt_frac]
        --threads &lt;INTEGER&gt;                  Number of parallel threads to use [default: 1]
        --valid-chars &lt;valid_chars&gt;          Valid characters in an alternative haplotype. This prevents non sequence-resolved
                                             variants from being genotyped. [default: ATGCatgc]
    -v, --vcf &lt;FILE&gt;                         Called variant file (VCF)</code></pre>
<h3 id="测试代码"><a href="#测试代码" class="headerlink" title="测试代码"></a>测试代码</h3><pre><code class="shell">./vartrix_linux -v ./test_dna.vcf -b ./test_dna.bam -f ./test_dna.fa -c ./dna_barcodes.tsv -o ./matrix.mtx</code></pre>
<p>vartrix的输出文件格式与Cell Ranger类似，</p>
<ul>
<li>第1列：VCF中的SNP所在行号</li>
<li>第2列：barcode在barcodes.tsv中的行号</li>
<li>第3列：相应barcode（细胞）SNP鉴定结果</li>
</ul>
<p>其中第3列的值只有<code>0，1，2，3</code>这四种，分别代表：</p>
<ul>
<li>0 : No Call</li>
<li>1 : ref/ref</li>
<li>2 : alt/alt</li>
<li>3 : alt/ref</li>
</ul>
<pre><code># head(./matrix.mtx)
%%MatrixMarket matrix coordinate real general
% written by sprs
46 1331 1073
1 8 1
1 73 1
1 154 2
1 174 1
1 191 2
1 194 2
1 243 1</code></pre>
<h2 id="UMAP图中展示VarTrix结果"><a href="#UMAP图中展示VarTrix结果" class="headerlink" title="UMAP图中展示VarTrix结果"></a>UMAP图中展示VarTrix结果</h2><p>要想在各细胞亚群UMAP图中展示VarTrix结果，我们需要先用seurat对细胞做常规分群分析，然后把VarTrix结果作为MetaData添加给Seurat对象。</p>
<h3 id="VarTrix结果转为可添加给Seurat对象的MetaData"><a href="#VarTrix结果转为可添加给Seurat对象的MetaData" class="headerlink" title="VarTrix结果转为可添加给Seurat对象的MetaData"></a>VarTrix结果转为可添加给Seurat对象的MetaData</h3><p>variants文件转为类似于geneID的格式，相当于给每个变异一个唯一的编号（以染色体及变异具体位置命名）</p>
<pre><code class="shell">awk &#39;&#123;print $1,$2&#125;&#39; test_dna.vcf &gt; SNV.loci.txt
sed -i &#39;s/\s/:/g&#39; SNV.loci.txt </code></pre>
<p><code>head SNV.loci.txt</code></p>
<pre><code>#CHROM:POS
1:13116
1:13118
1:13656
1:15211
1:15274</code></pre>
<h4 id="读取VarTrix的结果matrix-barcodes和variants"><a href="#读取VarTrix的结果matrix-barcodes和variants" class="headerlink" title="读取VarTrix的结果matrix, barcodes和variants"></a>读取VarTrix的结果matrix, barcodes和variants</h4><pre><code class="R">library(Seurat)
library(Matrix)
library(stringr)

setwd(&quot;~/Documents/RD/Rs/VarTrix&quot;)
snv_matrix &lt;- readMM(&quot;matrix.mtx&quot;)

# convert the matrix to a dataframe
snv_matrix &lt;- as.data.frame(as.matrix(t(snv_matrix)))
#read in the cell barcodes output by Cell Ranger
barcodes &lt;- read.table(&quot;dna_barcodes.tsv&quot;, header = F)

# read in SNV loci
# Should be constructed a single column. For example

# chr1:1234-1235
# chr2:2345-2346

# Construct the final table to add to the Seurat object
snps &lt;- read.table(&quot;SNV.loci.txt&quot;, header = F)
colnames(snv_matrix) &lt;- barcodes$V1
row.names(snv_matrix) &lt;- snps$V1

snv_matrix_t &lt;- as.data.frame(t(snv_matrix))

snv_matrix_t &lt;- str_replace(as.character(snv_matrix_t), &quot;0&quot;, &quot;No Call&quot;)
snv_matrix_t &lt;- str_replace(as.character(snv_matrix_t), &quot;1&quot;, &quot;ref/ref&quot;)
snv_matrix_t &lt;- str_replace(as.character(snv_matrix_t), &quot;2&quot;, &quot;alt/alt&quot;)
snv_matrix_t &lt;- str_replace(as.character(snv_matrix_t), &quot;3&quot;, &quot;alt/ref&quot;)</code></pre>
<h4 id="添加VarTrix信息给Seurat对象"><a href="#添加VarTrix信息给Seurat对象" class="headerlink" title="添加VarTrix信息给Seurat对象"></a>添加VarTrix信息给Seurat对象</h4><pre><code class="R">Seurat.obj&lt;-readRDS(&quot;/path/Step5_combinedObj.rds&quot;)
Seurat.obj &lt;- AddMetaData(object = Seurat.obj, metadata = snv_matrix_t)</code></pre>
<p><strong>UMAP展示</strong></p>
<pre><code class="R">UMAPPlot(object = Seurat.obj, group.by = &quot;1:13116&quot;)</code></pre>
<h4 id="UMAP展示结果"><a href="#UMAP展示结果" class="headerlink" title="UMAP展示结果"></a>UMAP展示结果</h4><p>因为VarTrix的github并未提供基因表达矩阵，所以VarTrix结果在UMAP中的展示我们暂用模拟数据代替</p>
<p>下面是模拟VarTrix结果添加给Seurat对象及可视化代码。</p>
<pre><code class="R">Seurat.obj&lt;-readRDS(&quot;~/Desktop/blish_covid.seu.rds&quot;)
gts &lt;- data.frame(snp_0001=rep(0, times=length(names(Seurat.obj$nFeature_RNA))),snp_0002=rep(1, times=length(names(Seurat.obj$nFeature_RNA))))
rownames(gts) &lt;- names(Seurat.obj$nFeature_RNA)
gts[sample(rownames(gts),4000), &quot;snp_0001&quot;] &lt;- 1
gts[sample(rownames(gts),5000), &quot;snp_0001&quot;] &lt;- 2
gts[sample(rownames(gts),6000), &quot;snp_0001&quot;] &lt;- 3

gts$snp_0001 &lt;- str_replace(as.character(gts$snp_0001), &quot;0&quot;, &quot;No Call&quot;)
gts$snp_0001 &lt;- str_replace(as.character(gts$snp_0001), &quot;1&quot;, &quot;ref/ref&quot;)
gts$snp_0001 &lt;- str_replace(as.character(gts$snp_0001), &quot;2&quot;, &quot;alt/alt&quot;)
gts$snp_0001 &lt;- str_replace(as.character(gts$snp_0001), &quot;3&quot;, &quot;alt/ref&quot;)

Seurat.obj &lt;- AddMetaData(object = Seurat.obj, metadata = gts)

UMAPPlot(object = Seurat.obj, group.by = &quot;snp_0001&quot;)</code></pre>
<p><img src="https://gitee.com/anno-sc/blog_source/raw/master/figure/VarTrix/vartrix_3.png"></p>
<p>其中<code>blish_covid.seu.rds</code>来自下面这篇文章</p>
<blockquote>
<p>Wilk A J, Rustagi A, Zhao N Q, et al. A single-cell atlas of the peripheral immune response in patients with severe COVID-19[J]. Nature Medicine, 2020: 1-7.</p>
</blockquote>
<blockquote>
<p><code>2020是不平凡的一年，在年终来临之际，我们计划组织一次单细胞线上高级分析培训班，希望能够给您提供一点参考，我们将以上面这篇文章的数据为基础线索进行高级分析的代码重现</code>，另外还有<a target="_blank" rel="noopener" href="https://b23.tv/8VQi5j">marker gene可视化应用</a>代码的赠送。</p>
</blockquote>

      
       <hr><span style="font-style: italic;color: gray;"> 转载请注明来源，欢迎对文章中的引用来源进行考证，欢迎指出任何有错误或不够清晰的表达。可以在下面评论区评论 </span>
    </div>
</article>





    <div id="comments"></div>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.css">

<script type="text/javascript">
    $.getScript('/js/gitalk.js', function () {
        var gitalk = new Gitalk({
            clientID: '9fea4bfda0aa0990cae3',
            clientSecret: 'a49f95594367f1707d1a9de1062630c5b16d7c86',
            repo: 'anno-sc.github.io',
            owner: 'seqyuan',
            admin: ['seqyuan'],
            id: decodeURI(location.pathname),
            distractionFreeMode: 'true',
            language: 'zh-CN',
            perPage: parseInt('10',10)
        })
        gitalk.render('comments')
    })
</script>




    




    </div>
    <div class="copyright">
        <p class="footer-entry">
    ©2020 anno-sc
</p>

    </div>
    <div class="full-toc">
        <button class="full" data-title="切换全屏 快捷键 s"><span class="min "></span></button>
<a class="" id="rocket" ></a>

    </div>
</div>

</body>
<script src="/js/jquery.pjax.js?v=1.1.0" ></script>

<script src="/js/script.js?v=1.1.0" ></script>
<script>
    var img_resize = 'default';
    function initArticle() {
        /*渲染对应的表格样式*/
        
            $("#post .pjax table").addClass("green_title");
        

        /*渲染打赏样式*/
        

        /*高亮代码块行号*/
        

        /*访问数量*/
        
        $.getScript("//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js");
        

        /*代码高亮，行号对齐*/
        $('.pre-numbering').css('line-height',$('.has-numbering').css('line-height'));

        
        
    }

    /*打赏页面隐藏与展示*/
    

</script>

<!--加入行号的高亮代码块样式-->

<!--自定义样式设置-->
<style>
    
    
    .nav {
        width: 542px;
    }
    .nav.fullscreen {
        margin-left: -542px;
    }
    .nav-left {
        width: 120px;
    }
    
    
    @media screen and (max-width: 1468px) {
        .nav {
            width: 492px;
        }
        .nav.fullscreen {
            margin-left: -492px;
        }
        .nav-left {
            width: 100px;
        }
    }
    
    
    @media screen and (max-width: 1024px) {
        .nav {
            width: 492px;
            margin-left: -492px
        }
        .nav.fullscreen {
            margin-left: 0;
        }
    }
    
    @media screen and (max-width: 426px) {
        .nav {
            width: 100%;
        }
        .nav-left {
            width: 100%;
        }
    }
    
    
    .nav-right .title-list nav a .post-title, .nav-right .title-list #local-search-result a .post-title {
        color: #383636;
    }
    
    
    .nav-right .title-list nav a .post-date, .nav-right .title-list #local-search-result a .post-date {
        color: #5e5e5f;
    }
    
    
    .nav-right nav a.hover, #local-search-result a.hover{
        background-color: #e2e0e0;
    }
    
    

    /*列表样式*/
    

    /* 背景图样式 */
    
    


    /*引用块样式*/
    

    /*文章列表背景图*/
    

    
</style>







</html>
